File: Exercise1/Exercise1.java
// liberia para leer por entrada
import java.util.Scanner;


public class ParOrImpar {
    // Main principal
    public static void main(String[] args) {

        // Necesitamos crear el objeto scanner
        Scanner scanner = new Scanner(System.in);

        // Leemos el numero por entrada
        System.out.print("Insert the number: ");
        int number = scanner.nextInt();

        // Procedemso a comprobar
        if(number % 2 == 0) {
            system.out.println("The descending numbers " + number)
            for (int i = number; i >= 0; i -= 2 ){
                System.out.println(i + " ");
            }
        } else {
            system.out.println("The descending numbers " + number)
            for (int i = number; i >= 1; i -= 1 ){
                System.out.println(i + " ");
            }
        }

        // cerramos el objeto scanner
        scanner.close();
    }
}

File: Exercise1/Exercise1.cpp
// == Ejercicio 1
/* Escriba un algoritmo que lea un número entero y determine si es par o impar. Si es par,
que escriba todos los pares de manera descendiente desde sí mismo y hasta el cero. Si
es impar, que escriba todos los impares de manera descendiente desde si sí mismo
hasta el uno. Utilice la instrucción LEER NUMERO al inicio del programa para cargar un
número en la variable NUMERO.*/


#include <iostream>
#include <sstream>

using namespace std;


int main (void) {
    
    // Variable en la que guardaremos el valor por entrada
    int number;

    // Leemos el valor por entrada
    cout << "Insert the number"
    cin >> number

    if (number % 2 == 0) {
        // Procedemos a mostrar todos los numeros pares desde el numero introducido
        cout << "The descending numbers " << number << "hasta 0\n"
        for (int i = number; i >= 0; i -= 2 ){
            cout << i << " ";
        }
    } else {
        //  Procedemos a mostrar todos los numeros impares 
        cout << "The descending numbers " << number << "hasta 1\n"
        for (int i = number; i >= 1; i -= 2 ){
            cout << i << " ";
        }
    }

    return 0;
}